x-pack/solutions/search/plugins/search_getting_started/public/components/header/add_data_button.tsx

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { useState } from 'react';
+import { EuiButton, EuiContextMenuPanel, EuiContextMenuItem, EuiPopover } from '@elastic/eui';
+
+import { i18n } from '@kbn/i18n';
+import { useKibana } from '../../hooks/use_kibana';
+
+export const AddDataButton: React.FC = () => {
+  const {
+    services: { application },
+  } = useKibana();
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+
+  const onButtonClick = () => {
+    setIsPopoverOpen(!isPopoverOpen);
+  };
+
+  const closePopover = () => {
+    setIsPopoverOpen(false);
+  };
+
+  const items = [
+    <EuiContextMenuItem
+      key="upload"
+      icon="exportAction"
+      onClick={() => {
+        closePopover();
```

Comment:
do we need this line ? the line below would navigate to another app. You could close popover  directly in `EuiPopover` 
```
<EuiPopover ... closePopover={() => setIsPopoverOpen(false)} />
```

Reply:
I think we do need this as `EuiPopover`'s `closePopover` prop just tells the popover how to close. Clicking a child `EuiContextMenuItem` doesn't automatically call it.